test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskLogicTests.cs

```diff
@@ -64,6 +65,137 @@ public void PackTaskLogic_ProducesBasicPackage()
+
+                // Act
+                tc.BuildPackage();
+
+                // Assert
+                var logger = (TestLogger)tc.Request.Logger;
+                var messages = logger.WarningMessages.ToArray();
+                Assert.True(messages[0].Contains("lib/net50/def.dll"));
+                Assert.True(messages[0].Contains("include dot"));
+            }
+        }
+
+        [Fact]
+        public void PackTaskLogic_WarnsMissingDot_UsingNuspec()
+        {
+            // Arrange
+            using (var testDir = TestDirectory.Create())
+            {
+                // Arrange
+                string nuspec = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<package >
+  <metadata>
+    <id>bar</id>
+    <version>0.0.0</version>
+    <title>bartitle</title>
+    <authors>kat</authors>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <license type=""expression"">MIT</license>
+    <description>desc</description>
+    <releaseNotes>release notes</releaseNotes>
+    <copyright>msft</copyright>
+    <tags>foo bar</tags>
+    <dependencies>
+        <group targetFramework=""net50"">
+            <dependency id=""Newtonsoft.Json"" version=""12.0.3""/>
+        </group>
+    </dependencies>
+  </metadata>
+</package>";
+                string nuspecPath = Path.Combine(testDir, "bar.nuspec");
+                File.WriteAllText(nuspecPath, nuspec);
+
+                var tc = new TestContext(testDir);
+                tc.Request.NuspecFile = nuspecPath;
+                tc.Request.NuspecBasePath = testDir;
+
+                var net50WinDllDir = Path.Combine(testDir, "lib", "net50");
+                var net50WinDllPath = Path.Combine(net50WinDllDir, "a.dll");
+
+                Directory.CreateDirectory(net50WinDllDir);
+                File.WriteAllBytes(net50WinDllPath, new byte[0]);
+
+                // Act
+                tc.BuildPackage();
+
+                // Assert
+                var logger = (TestLogger)tc.Request.Logger;
+                var messages = logger.WarningMessages.ToArray();
+                Assert.True(messages[0].Contains("net50"));
+                Assert.True(messages[0].Contains("include dots"));
+            }
+        }
+
+        [Fact]
+        public void PackTaskLogic_WhenDotInPlatformOnly_WarnsMissingDot_UsingNuspec()
+        {
+            // Arrange
+            using (var testDir = TestDirectory.Create())
+            {
+                // Arrange
+                string nuspec = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<package >
+  <metadata>
+    <id>bar</id>
+    <version>0.0.0</version>
+    <title>bartitle</title>
+    <authors>kat</authors>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <license type=""expression"">MIT</license>
+    <description>desc</description>
+    <releaseNotes>release notes</releaseNotes>
+    <copyright>msft</copyright>
+    <tags>foo bar</tags>
+    <dependencies>
+        <group targetFramework=""net50-windows7.0"">
```

Comment:
like the other test, it's not clear if this test is supposed to be testing TFMs in the nuspec or not. Historically it's been valid for a nupkg to not have any dependencies. As of last year, it will generate a warning, so your assertions can take that into account and ignore that warning, and check only for NU5501. That makes it more clear, in my opinion, what this test is actually testing.